src/mutable_arithmetics.jl

```diff
@@ -286,7 +286,11 @@ end
 function _MA.add_mul(lhs::AbstractJuMPScalar, x::_Scalar, y::_Scalar)
     T = _MA.promote_operation(_MA.add_mul, typeof(lhs), typeof(x), typeof(y))
     expr = _MA.operate(convert, T, lhs)
-    return _MA.operate!(_MA.add_mul, expr, x, y)
+    # We can't use `operate!!` here because that will cause a StackOverflow.
```

Comment:
If that's a bug in MA, we should open an issue

Reply:
I don't think it's a bug, it's expected behavior.

I updated the comment. Basically, if `IsNotMutable`, then `operate!!` will fallback to calling this `add_mul` method and cause an infinite recursion (partially also because `expr` is the same type as `lhs`).